test/e2e/serial/tests/operator_profile_update.go

```diff
@@ -0,0 +1,282 @@
+		// and currently is missing and likely to be addressed in https://issues.redhat.com/browse/CNF-10552.
+		var testPod *corev1.Pod
+		nropObjInitial := &nropv1.NUMAResourcesOperator{}
+		nroKey := objects.NROObjectKey()
+
+		BeforeEach(func() {
+			By("disable RTE functionality hence publishing updated NRT data")
+			err := fxt.Client.Get(context.TODO(), nroKey, nropObjInitial)
+			Expect(err).ToNot(HaveOccurred(), "cannot get %q from the cluster", nroKey.String())
+
+			updatedNROPObj := nropObjInitial.DeepCopy()
+			infoRefreshPauseMode := nropv1.InfoRefreshPauseEnabled
+			//TODO adapt the existence of multiple node groups,https://issues.redhat.com/browse/CNF-10552
+			updatedNROPObj.Spec.NodeGroups[0].Config = &nropv1.NodeGroupConfig{
+				InfoRefreshPause: &infoRefreshPauseMode,
+			}
+
+			err = fxt.Client.Update(context.TODO(), updatedNROPObj)
+			Expect(err).ToNot(HaveOccurred())
+			klog.Info("wait long enough to verify NROP object is updated")
+			Eventually(func() nropv1.InfoRefreshPauseMode {
+				nropObjCurrent := &nropv1.NUMAResourcesOperator{}
+				err = fxt.Client.Get(context.TODO(), nroKey, nropObjCurrent)
+				Expect(err).ToNot(HaveOccurred(), "post NROP updated: cannot get %q from the cluster", nroKey.String())
+				//TODO adapt the existence of multiple node groups,https://issues.redhat.com/browse/CNF-10552
+				currentMode := nropObjCurrent.Status.MachineConfigPools[0].Config.InfoRefreshPause
+				if currentMode != &infoRefreshPauseMode {
+					klog.Warningf("nodegroup status is not completely updated yet. Expecting InfoRefreshPause to equal %q but found %q", &infoRefreshPauseMode, currentMode)
+				}
+				return *currentMode
+			}).WithTimeout(2*time.Minute).WithPolling(9*time.Second).Should(Equal(infoRefreshPauseMode), "failed to update the NROP object")
+
+			klog.Info("wait for the ds to get its args updated")
+			dsKey := wait.ObjectKey{
+				Namespace: nropObjInitial.Status.DaemonSets[0].Namespace,
+				Name:      nropObjInitial.Status.DaemonSets[0].Name,
+			}
+			Eventually(func() []string {
+				dsObj := appsv1.DaemonSet{}
+				err = fxt.Client.Get(context.TODO(), client.ObjectKey(dsKey), &dsObj)
+				Expect(err).ToNot(HaveOccurred())
+				cnt := k8swgobjupdate.FindContainerByName(dsObj.Spec.Template.Spec.Containers, manifests.ContainerNameRTE)
+				Expect(cnt).NotTo(BeNil(), "cannot find container data for %q", manifests.ContainerNameRTE)
+				return cnt.Args
+
+			}).WithTimeout(time.Minute).WithPolling(9*time.Second).Should(ContainElement(ContainSubstring("--no-publish")), "ds was not updated as expected: \"--no-pubish\" arg is missing.")
+			klog.Info("waiting for DaemonSet to be ready")
+			_, err = wait.With(e2eclient.Client).Interval(10*time.Second).Timeout(3*time.Minute).ForDaemonSetReadyByKey(context.TODO(), dsKey)
+			Expect(err).ToNot(HaveOccurred(), "failed to get the daemonset %s: %v", dsKey.String(), err)
+		})
+
+		AfterEach(func() {
+			restoreNROP(fxt, *nropObjInitial)
```

Comment:
if a function is called only in a single place, please inline it. Let's consider extracting functions only when we have at least two call sites.

Reply:
agree. this will be used in later multi-staged tests. This is why I extracted it to a method.